From 43f627fc36b5a958509f438fcdc621be33763163 Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 09:21:12 +0000
Subject: [PATCH] [wasm][test] Prefer -sdk instead of -Xcc --sysroot

---
 test/lit.cfg | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/test/lit.cfg b/test/lit.cfg
index 2d73066157b..c2931990af7 100644
--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -1853,9 +1853,8 @@ elif run_os == 'wasi':
     config.target_build_swift = ' '.join([
         config.swiftc,
         '-target', config.variant_triple,
-        '-Xcc', '--sysroot=%s' % config.variant_sdk,
-        '-Xclang-linker', '--sysroot=%s' % config.variant_sdk,
-        '-toolchain-stdlib-rpath', config.resource_dir_opt,
+        '-sdk', config.variant_sdk,
+        '-toolchain-stdlib-rpath', config.resource_dir_opt, 
         mcp_opt, config.swift_test_options,
         config.swift_driver_test_options, swift_execution_tests_extra_flags])
     config.target_codesign = "echo"
@@ -1866,7 +1865,7 @@ elif run_os == 'wasi':
     config.target_swift_frontend = ' '.join([
         config.swift_frontend,
         '-target', config.variant_triple,
-        '-Xcc', '--sysroot=%s' % config.variant_sdk,
+        '-sdk', config.variant_sdk,
         config.resource_dir_opt, mcp_opt,
         config.swift_test_options, config.swift_frontend_test_options])
     subst_target_swift_frontend_mock_sdk = config.target_swift_frontend
-- 
2.40.0

