From a858c9d561f1c2a8f2bd94bb0ff31ac16f4bf0db Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Thu, 29 Jun 2023 05:57:18 +0000
Subject: [PATCH] [test] Make KeyPath test pointer-size agnostic part 2

---
 test/stdlib/KeyPath.swift | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/stdlib/KeyPath.swift b/test/stdlib/KeyPath.swift
index 1358345f2e2..4e929110299 100644
--- a/test/stdlib/KeyPath.swift
+++ b/test/stdlib/KeyPath.swift
@@ -1097,7 +1097,7 @@ if #available(SwiftStdlib 5.9, *) {
     let catNameKp = _createOffsetBasedKeyPath(
       root: Cat.self,
       value: String.self,
-      offset: 16
+      offset: 2 * MemoryLayout<UnsafeRawPointer>.size
     ) as? KeyPath<Cat, String>
 
     expectNotNil(catNameKp)
-- 
2.40.0

