From a885d4bbc2ddb919311054153f6f0bd3bde8c6fb Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Fri, 20 Sep 2024 08:06:15 +0900
Subject: [PATCH] CMake: Fix accidental variable expansion of `WASI` (#45)

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f371b1..e16d54b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,7 +57,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL Windows)
         $<$<COMPILE_LANGUAGE:C,CXX>:U_TIMEZONE=_timezone>
         $<$<COMPILE_LANGUAGE:C,CXX>:_CRT_SECURE_NO_DEPRECATE>
         $<$<COMPILE_LANGUAGE:C,CXX>:U_PLATFORM_USES_ONLY_WIN32_API>)
-elseif(CMAKE_SYSTEM_NAME STREQUAL WASI)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "WASI")
     # WASI specific settings
     add_compile_definitions(
         $<$<COMPILE_LANGUAGE:C,CXX>:U_HAVE_TZSET=0>
-- 
2.46.0

