From 98a188879dae33eccadfab1bd661c1d03f0e1bc7 Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 09:27:38 +0000
Subject: [PATCH] [wasm][test] HACK: Disable os_stdlib tests due to no dynamic
 loading

---
 test/lit.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/lit.cfg b/test/lit.cfg
index 80abad2fb6a..ee26afc1a50 100644
--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -2286,7 +2286,7 @@ if run_vendor != 'apple':
 
 if 'remote_run_host' in lit_config.params:
     configure_remote_run()
-elif not kIsWindows:
+elif not kIsWindows and not run_os == 'wasi':
     if 'use_os_stdlib' in lit_config.params:
         config.available_features.add('use_os_stdlib')
 
-- 
2.40.0

