From 6ed145de5bf0c46f48e6057ea26e91e58dc32c2c Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 09:08:40 +0000
Subject: [PATCH] [wasm][test] Add test case for WebAssembly object file format

---
 test/DebugInfo/modulecache.swift | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/DebugInfo/modulecache.swift b/test/DebugInfo/modulecache.swift
index 1dceafbf15d..22d126eca7c 100644
--- a/test/DebugInfo/modulecache.swift
+++ b/test/DebugInfo/modulecache.swift
@@ -16,7 +16,7 @@ import ClangModule
 // RUN: %empty-directory(%t)
 // RUN: %target-swift-frontend %s -c -g -o %t.o -module-cache-path %t -I %S/Inputs
 // RUN: llvm-readobj -h %t/*/ClangModule-*.pcm | %FileCheck %s
-// CHECK: Format: {{(Mach-O|ELF|elf64|COFF|elf32-littlearm)}}
+// CHECK: Format: {{(Mach-O|ELF|elf64|COFF|elf32-littlearm|WASM)}}
 
 // 3. Test that swift-ide-check will not share swiftc's module cache.
 
-- 
2.40.0

