From 2c1e67d9a0582b2bf8dbb3726c8873fcaeefcc69 Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Thu, 29 Jun 2023 05:57:18 +0000
Subject: [PATCH] [test] Make KeyPath test pointer-size agnostic part 2

---
 test/stdlib/KeyPath.swift | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/stdlib/KeyPath.swift b/test/stdlib/KeyPath.swift
index 47a30a15007..59a43a44bbf 100644
--- a/test/stdlib/KeyPath.swift
+++ b/test/stdlib/KeyPath.swift
@@ -1100,7 +1100,7 @@ if #available(SwiftStdlib 5.9, *) {
     let catNameKp = _createOffsetBasedKeyPath(
       root: Cat.self,
       value: String.self,
-      offset: 16
+      offset: 2 * MemoryLayout<UnsafeRawPointer>.size
     ) as? KeyPath<Cat, String>
 
     expectNotNil(catNameKp)
-- 
2.40.0

