From af251aefeb5ec75b226ab4443491971f08f0528e Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Tue, 20 Jun 2023 09:24:47 +0000
Subject: [PATCH] [wasm][test] Add missing target specific lit properties

---
 test/lit.cfg | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/test/lit.cfg b/test/lit.cfg
index 89b8137012a..3c4aa7f0c2e 100644
--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -1876,10 +1876,16 @@ elif run_os == 'wasi':
     config.target_sil_opt = (
         '%s -target %s %s %s %s' %
         (config.sil_opt, config.variant_triple, config.resource_dir_opt, mcp_opt, config.sil_test_options))
+    subst_target_sil_opt_mock_sdk = config.target_sil_opt
+    subst_target_sil_opt_mock_sdk_after = ""
     config.target_swift_symbolgraph_extract = ' '.join([
         config.swift_symbolgraph_extract,
         '-target', config.variant_triple,
         mcp_opt])
+    config.target_swift_api_extract = ' '.join([
+        config.swift_api_extract,
+        '-target', config.variant_triple,
+        '-sdk', shell_quote(config.variant_sdk)])
     config.target_swift_ide_test = (
         '%s -target %s %s %s %s %s' %
         (config.swift_ide_test, config.variant_triple, config.resource_dir_opt,
-- 
2.40.0

